packages/protocol/test/authorizers/Api3TokenLock.sol.js

```diff
@@ -0,0 +1,755 @@
+          });
+        });
+      });
+
+      context('caller is airnodeAdmin', async () => {
+        context('requester is being blocked on airnodeAdmin`s airnode', async () => {
+          it('sets the block', async () => {
+            await api3TokenLock.connect(airnodeWallet).blockRequester(airnodeAddress, roles.requester.address);
+
+            expect(
+              await api3TokenLock.airnodeToRequesterToBlockStatus(airnodeAddress, roles.requester.address)
+            ).to.equal(true);
+
+            await expect(
+              api3TokenLock.connect(roles.sponsor).lock(airnodeAddress, roles.requester.address)
+            ).to.revertedWith(Errors.RequesterBlocked);
+
+            await api3TokenLock.connect(roles.sponsor).lock(roles.unknown.address, roles.requester.address);
+            const now = (await hre.ethers.provider.getBlock(await hre.ethers.provider.getBlockNumber())).timestamp;
+            const expectedUnlockTime = now + ONE_MINUTE_IN_SECONDS;
+
+            expect(
+              await api3TokenLock.sponsorLockAmount(
+                roles.unknown.address,
+                roles.requester.address,
+                roles.sponsor.address
+              )
+            ).to.equal(LOCK_AMOUNT);
+            expect(
+              await api3TokenLock.sponsorUnlockTime(
+                roles.unknown.address,
+                roles.requester.address,
+                roles.sponsor.address
+              )
+            ).to.equal(expectedUnlockTime);
+          });
+
+          it('emits event', async () => {
+            await expect(api3TokenLock.connect(airnodeWallet).blockRequester(airnodeAddress, roles.requester.address))
+              .to.emit(api3TokenLock, 'BlockedRequester')
+              .withArgs(airnodeAddress, roles.requester.address, airnodeWallet.address);
+          });
+        });
+        context('requester is being blocked on another airnode', async () => {
+          it('reverts', async () => {
+            await expect(
+              api3TokenLock.connect(airnodeWallet).blockRequester(roles.unknown.address, roles.requester.address)
+            ).to.revertedWith(Errors.LowRank);
+          });
+        });
+        context('requester is being blocked globally', async () => {
+          it('reverts', async () => {
+            await expect(
+              api3TokenLock
+                .connect(airnodeWallet)
+                .blockRequester(hre.ethers.constants.AddressZero, roles.requester.address)
+            ).to.revertedWith(Errors.LowRank);
+          });
+        });
+      });
+      context('caller is unknown', async () => {
+        it('reverts', async () => {
+          await expect(
+            api3TokenLock.connect(roles.unknown).blockRequester(airnodeAddress, roles.requester.address)
+          ).to.revertedWith(Errors.LowRank);
+        });
+      });
+    });
+
+    describe('lock', async () => {
```

Comment:
Should we also add at least one happy path were we sponsor a single requester with multiple sponsors?

Reply:
isn't [this](https://github.com/api3dao/airnode/blob/3137035fbabbb8967323cde9daec5f78e7b6b449/packages/protocol/test/authorizers/Api3TokenLock.sol.js#L612) what you're referring to?

Reply:
You are right, my bad. I got confused because it was under 'caller has already locked' context and I was expecting that one to revert.